### PR TITLE
[DO NOT MERGE] Vocabulary builder in deepchem

### PR DESCRIPTION
**Pull request by gusty1g**
# Pull Request Template

## Description

Adding a vocabulary builder in DeepChem which can be used for downstream pretraining tasks.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/vocabulary_builders/vocabulary_builder.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+from deepchem.data import Dataset
+
+
+class VocabularyBuilder():
```
Comment (rbharath) [inline deepchem/feat/vocabulary_builders/vocabulary_builder.py:4]: I think this basic API looks reasonable

File: deepchem/feat/huggingface_featurizer.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from deepchem.feat.base_classes import Featurizer
+
+
+class HuggingFaceFeaturizer(Featurizer):
```
Comment (rbharath) [inline deepchem/feat/huggingface_featurizer.py:4]: This looks useful. Having the ability to use arbitrary huggingface tokenizers will be useful for DeepChem model building and chemberta

**Review by rbharath — COMMENTED** [review]
Basic design here looks good to proceed. Can you update the forum post with a link to this draft in case other community members have feedback as well?

File: deepchem/feat/vocabulary_builders/vocabulary_builder.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/feat/vocabulary_builders/vocabulary_builder.py:4]: I will break this into 4 subsequent pull requests and add them one at a time - base vocabulary builder (#3265), hugging face tokenizer, hugging face vocabulary builder and lastly, grover.

**Pull request closed without merging**